src/bigint.rs

```diff
@@ -811,6 +829,28 @@ impl Signed for BigInt {
     }
 }
 
+impl<T: PrimInt> Pow<T> for BigInt
+    where BigUint: Pow<T, Output=BigUint>
```

Comment:
Let's have exponent references too, as I mentioned on `BigUint`.

I think it's *technically* a breaking change to do this generically, as someone could already have an impl, however silly, like:

```rust
struct Foo(usize);

impl Pow<Foo> for BigInt {
    type Output = BigInt;
    
    fn pow(self, exp: Foo) -> BigInt {
        num_traits::pow(self, exp.0)
    }
}
```

Let's just avoid that question by implementing it in macros like `BigUint`.